lesson_21/homework_21.py

```diff
@@ -0,0 +1,41 @@
+from datetime import datetime, timedelta
+from pathlib import Path
+
+from lesson_21.logger import logger
+
+DIR_PATH = Path(__file__).parent / 'hblog.txt'
+TARGET_KEY = 'Key TSTFEED0300|7E3E|0400'
+
+
+def get_log(path):
+    with open(path, 'r', encoding='utf') as f:
+        lines = f.readlines()
+        for line in lines:
+            yield line
+
+
+def analyze_heartbeat(path, target_key):
+    logs = get_log(path)
+    timestamps = []
+
+    data_log = [log for log in logs if target_key in log]
+
+    for log in data_log:
+        pos = log.find("Timestamp ")
+        if pos != -1:
+            timestamp = log[pos + len("Timestamp "):pos + len("Timestamp ") + 8]
+            timestamps.append(datetime.strptime(timestamp, '%H:%M:%S'))
```

Comment:
```suggestion
            timestamps.append(datetime.strptime(timestamp, '%H:%M:%S'))
```
проігнорували ви дату, але в лозі можуть буть дані за декілька днів, і мати дату було б добое

Reply:
в файле нет данных о датах